src/lib/comms/include/sol-connman.h

```diff
@@ -0,0 +1,285 @@
+
+/**
+ * @brief Gets the service state
+ *
+ * This function gets the state for the connman service,
+ * after the service is to give by service monitor.
+ *
+ * @note SOL_CONNMAN_SERVICE_STATE_REMOVE is used to show
+ * the service has been removed. When it is notified,
+ * the service has been removed in the system.
+ *
+ * @param service The service structure which the state is desired
+ *
+ * @return The state of the services on success,
+ * SOL_CONNMAN_SERVICE_STATE_UNKNOWN on failure.
+ */
+enum sol_connman_service_state sol_connman_service_get_state(
+    const struct sol_connman_service *service);
+
+/**
+ * @brief Gets the service type
+ *
+ * This function gets the type for the connman service,
+ * after the service is to give by service monitor.
+ *
+ * @param service The service structure which the type is desired
+ *
+ * @return The type of the services on success,
+ * SOL_CONNMAN_SERVICE_TYPE_UNKNOWN on failure.
+ */
+const char *sol_connman_service_get_type(
+    const struct sol_connman_service *service);
+
+/**
+ * @brief Gets the service network address
+ *
+ * This function gets the network address for the connman service,
+ * after the service is to give by service monitor.
+ *
+ * @param service The service structure which the network address is desired
+ * @param family which type of the network address is desired (ipv4 or ipv6).
+ *
+ * @return A point to sol_network_link_addr created on success, NULL on failure.
+ */
+struct sol_network_link_addr *sol_connman_service_get_network_address(
+    const struct sol_connman_service *service, enum sol_network_family family);
+
+/**
+ * @brief Gets the service strength
+ *
+ * This function gets the service strength for the connman service,
+ * after the service is to give by service monitor.
+ *
+ * @param service The service structure which the service strength is
+ * desired
+ *
+ * @return 0-100 on success, -errno on failure.
+ */
+int32_t sol_connman_service_get_strength(
+    const struct sol_connman_service *service);
+
+/**
+ * @brief Gets the call result for service
+ *
+ * This function gets the call result for service.
+ * Since the connman function is asynchronous, the call return is not connection error,
+ * but just some dispatching/immediate error. The actual state change will be notified
+ * via sol_connman_add_service_monitor() callbacks.
+ * After the actual state change is notified via sol_connman_add_service_monitor()
+ * callbacks, sol_connman_service_get_call_result can be used to get the actual state.
+ *
+ * @return TRUE on success, FALSE on failure
+ */
+bool sol_connman_service_get_call_result(
+    const struct sol_connman_service *service);
+
+/**
+ * @brief Gets the global connection state of system
+ *
+ * This function gets the global connection state for system.
+ *
+ * @return The state of the global connection state on success,
+ * SOL_CONNMAN_STATE_UNKNOWN on failure
+ */
+enum sol_connman_state sol_connman_get_state(void);
```

Comment:
we need a monitor for the base properties (State, Offline...). We can do a single monitor that says "something changed", and then you can get the new value if you're interested in it. Then you should also refer `@see sol_connman_...` and say that the pattern is to add the monitor BEFORE call get_state()/get_offline()...